xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java

```diff
@@ -223,20 +223,15 @@ public void endLink(ResourceReference reference, boolean freestanding, Map<Strin
         }
     }
 
-    @Override
+        @Override
     public void beginFormat(Format format, Map<String, String> parameters)
     {
         // If the previous format had parameters and the parameters are different from the current ones then close them
         if (this.previousFormatParameters != null) {
-            if (parameters.isEmpty()) {
-                // print("(%%)");
-                // this.previousFormatParameters = null;
-            } else if (!this.previousFormatParameters.equals(parameters)) {
-                this.previousFormatParameters = null;
+            if (!this.previousFormatParameters.equals(parameters)) {
```

Comment:
If previousFormatParameters is empty you will enter this if while it used to not be the case before.

Reply:
Not sure what you mean here. Is something like this correct?
```

Reply:
I mean that after your change you won't have the same behavior if both previousFormatParameters and parameters are empty.